src/actions/composite.js

```diff
@@ -25,7 +25,13 @@ export const updateApp = (change, router) => dispatch => {
   dispatch(updateFilters(change))
 
   if (router) {
-    history.push(createNewLocation({ change, router }))
+    const loc = createNewLocation({ change, router })
+    if (window.caches) {
```

Comment:
can you explain this? what if `loc.pathname` is already in the cache - will it override or append the old one?